### ISSUE TITLE
Dependencies referenced by path.

### ISSUE THREAD

**Issue by jD91mZM2** [japaric/cross#139]
Example:
```
crate = { path = "../crate" }
```

This fails with
```
error: failed to load source for a dependency on `crate`

Caused by:
  Unable to update file:///crate

Caused by:
  failed to read `/crate/Cargo.toml`

Caused by:
  No such file or directory (os error 2)
```

**Comment by japaric** [comment]
This is expected and documented in the [README](https://github.com/japaric/cross#caveats--gotchas).

**Comment by jD91mZM2** [comment]
Sorry, my bad

**Comment by gnzlbg** [comment]
If I have the following structure:

* `root_crate/`
  * `src/`
  * `examples/`
    * `example_crate/`
        * `src/`

where `example_crate` uses `path = ../../` to refer to the root crate, and one executes from the `root_crate`: `cross test -p examples/example_crate` this should work though as long as `example_crate` is part of the `root_crate`'s workspace. 

However, if one excludes `example_crate` from the `root_crate`'s workspace, then doing `cross test --manifest-path examples/example_crate` will fail, because that will try to write a `Cargo.lock` into the `examples/example_crate` directory, which is read only.

**Comment by Alexhuszagh** [comment]
This was fully implemented in #684.
